Libraries/Network/RCTNetworking.m

```diff
@@ -209,7 +209,10 @@ - (RCTURLRequestCancellationBlock)buildRequest:(NSDictionary<NSString *, id> *)q
   NSURL *URL = [RCTConvert NSURL:query[@"url"]]; // this is marked as nullable in JS, but should not be null
   NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:URL];
   request.HTTPMethod = [RCTConvert NSString:RCTNilIfNull(query[@"method"])].uppercaseString ?: @"GET";
-  request.allHTTPHeaderFields = [RCTConvert NSDictionary:query[@"headers"]];
+  NSMutableDictionary *headers = [[RCTConvert NSDictionary:query[@"headers"]] mutableCopy];
+  [headers enumerateKeysAndObjectsUsingBlock:^(NSString *  _Nonnull key, id  _Nonnull obj, BOOL * _Nonnull stop) {
+    [headers setValue:[NSString stringWithFormat:@"%@", obj] forKey:key];
```

Comment:
A couple of issues with this:

1) `setValue:forKey:` is actually part of the KVC (Key Value Coding) protocol, and should not be used for setting dictionary keys. The method you probably meant was `setObject:forKey:`, but we prefer to use `dict[key] = value` syntax anyway. I'd write it as:

    headers[key] = [value description];

2) Are you sure it's safe to mutate the `headers` dictionary while you are enumerating it? That normally crashes with a "Collection was mutated while being enumerated." exception.